### ISSUE TITLE
Quotation marks in host vars breaks inventory file

### ISSUE THREAD

**Issue by theasp** [ansible/awx#546]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - UI

##### SUMMARY

I have a dynamic inventory script that used to work in AWX when it was using Ansible 2.3, but now I get the following warning, and no tasks are executed, *BUT* AWX shows the job as successful.

```
 [WARNING]:  * Failed to parse /tmp/awx_318_V8mVNi/inventory with yaml plugin:
Syntax Error while loading YAML.   The error appears to have been in
'/tmp/awx_318_V8mVNi/inventory': line 3, column 718, but may be elsewhere in
the file depending on the exact syntax problem.  The offending line appears to
be:
[...]
```

The output of my dynamic script is actually ok, but the script that AWX is generating outputs invalid JSON when ran:

```
[root@awx x]# ./awx_314_QzQbYO/inventory > /dev/null
[root@awx x]# echo $?
0
[root@awx x]# ./awx_314_QzQbYO/inventory | jq .
parse error: Invalid numeric literal at line 1, column 76884
```

After some digging, I found that the quotations inside of a string aren't being escaped when AWX writes out the inventory script:
```json
[...]
 "_meta":
  "host1":
   {"ansible_ssh_common_args": "-o ProxyCommand="ssh -W %h:%p -q user@host2""}
```
The input inventory script outputs:
```json
        "ansible_ssh_common_args": "-o ProxyCommand=\"ssh -W %h:%p -q user@host2\""
```

##### ENVIRONMENT
* AWX version: 1.0.1.120
* AWX install method: docker
* Ansible version:  2.4.1.0 
* Operating System: Ubuntu 16.04
* Web Browser: Chrome

##### STEPS TO REPRODUCE

Add a host variable with quotations inside of it, then try to run a playbook.  This may require the result of a dynamic inventory script, I've not tried using the UI inventory.

##### EXPECTED RESULTS

Bacon to fall from the sky, or my playbook to run.  It also should have also marked the job as failed.

##### ACTUAL RESULTS

No bacon and my playbook doesn't run.  Job shows as succeeded in the UI.

**Comment by theasp** [comment]
Switching to single quotes inside my variable fixes the problem for me.

May I suggest generating the inventory script be written like this:
```bash
#!/bin/sh

cat <<"EOF"
${JSON_GOES_HERE}"
EOF
```
Rather than:
```python
#! /usr/bin/env python
# -*- coding: utf-8 -*-
"""<JSON>"""
```

**Comment by cchurch** [comment]
Looks like it's related to this recent change where we dump the JSON to a file to output: https://github.com/ansible/awx/pull/459/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R684

The escaped quotes in the JSON string (`\"`) are being interpreted by Python in parsing the string literal instead of being output as-is. I think just changing `print """%s"""` to `print %r` in that format string would fix it.

Let's get this guy some :bacon:!

**Comment by cchurch** [comment]
Fixed via #551.

![mmm bacon](https://media0.giphy.com/media/11sFH7QYAzLE0E/giphy.gif)
